### PR TITLE
Add Node.js reference implementation of Twt Hash algorithm

### LINKED ISSUE #361: Twt hash reference implementation in `node`
https://github.com/jointwt/twtxt/blob/f18dfa2efc5c8c209bbf68f6bab15c7144d6f1e1/docs/_posts/2020-12-11-twthashextension.md#L83-L85

Here is my modest contribution to twt hash algorithm implementations in form of a `node` powered Repl.it _notebook_: https://repl.it/@vladimyr1/hash-twt

It is a bit hard to reduce it to a simple snippet because it heavily relies on external libs:
```js
const b32encode = require('base32-encode');
const blake2 = require('blake2');
```

I guess something along these lines might be a good fit:
```js
const b32encode = require('base32-encode');
const blake2 = require('blake2');

function base32(payload) {
  return b32encode(Buffer.from(payload), 'RFC3548', { padding: false })
    .toLowerCase();
}

function blake2b256(payload) {
  return blake2.createHash('blake2b', { digestLength: 32 })
    .update(Buffer.from(payload))
    .digest();
}

const payload = [feedUrl, twt.createdAt, twt.content].join('\n');
const hash = base32(blake2b256(payload)).slice(-7);
```

What do you think, is it PR ready material? :upside_down_face:

### PR DESCRIPTION
**Pull request by vladimyr**
Closes #361 :tada:

### CODE REVIEW

**Review by prologic — APPROVED** [review]
LGTM! 🎉

@der-lyse for final once-over.

File: docs/_posts/2020-12-11-twthashextension.md [hunk 1]
```diff
@@ -96,10 +96,37 @@ hash = hash[len(hash)-7:]
...
+}
+
+function blake2b256(payload) {
+  return blake2.createHash('blake2b', { digestLength: 32 })
+    .update(Buffer.from(payload))
+    .digest();
+}
+
+function formatRFC3339(text) {
+  return DateTime.fromISO(text, { setZone: true, zone: 'utc' })
```
Comment (der-lyse) [inline docs/_posts/2020-12-11-twthashextension.md:124]: Just want to make sure: Does `zone: 'utc'` always convert to UTC or is it the default timezone, if the timestamp is missing one?

The original TZ must be preserved, if present. However, UTC must be assumed, when missing.

**Review by der-lyse — COMMENTED** [review]
I don't have a node setup, so I did not check whether this actually works. The code looks plausible, but I have one question. I verified that the docs still build. The syntax highlighter for both Go and Python kinda sucks, but for JS looks decent, so I reckon we can just keep it.

File: docs/_posts/2020-12-11-twthashextension.md (hunk 1 above)
Comment (vladimyr) [reply docs/_posts/2020-12-11-twthashextension.md:124]: Thanks for asking. That is just the default timezone fallback. 👍

**Comment by vladimyr** [comment]
> I don't have a node setup, so I did not check whether this actually works. 

Now you do: https://repl.it/@vladimyr1/hash-twt 😉

> I have one question. I verified that the docs still build. The syntax highlighter for both Go and Python kinda sucks, but for JS looks decent, so I reckon we can just keep it.

Ok, I'll remove the highlight then completely to keep things consistent.

File: docs/_posts/2020-12-11-twthashextension.md (hunk 1 above)
Comment (der-lyse) [reply docs/_posts/2020-12-11-twthashextension.md:124]: Alright, perfect! :+1:

**Comment by der-lyse** [comment]
> Now you do: https://repl.it/@vladimyr1/hash-twt wink

Hehe, thanks.

> Ok, I'll remove the highlight then completely to keep things consistent.

I think, we'll just go with it and see, how many people complain about it. ;-)

**Review by der-lyse — APPROVED** [review]

**Pull request merged**